webpack.config.development.js

```diff
@@ -79,11 +121,28 @@ export default validate(merge(baseConfig, {
      */
     new webpack.DefinePlugin({
       'process.env.NODE_ENV': JSON.stringify('development')
-    })
+    }),
+    // turn debug mode on.
+    new webpack.LoaderOptionsPlugin({
+      debug: true
+    }),
   ],
 
   /**
    * https://github.com/chentsulin/webpack-target-electron-renderer#how-this-module-works
    */
-  target: 'electron-renderer'
-}));
+  target: 'electron-renderer',
+  devServer: {
+    port,
+    hot: true,
+    inline: false,
+    historyApiFallback: true,
+    contentBase: path.join(__dirname, 'dist'),
+    publicPath,
+    setup() {
+      spawn('npm', ['run', 'start-hot'], { shell: true, env: process.env, stdio: 'inherit' })
```

Comment:
I don't think that will work. If you try passing `npm run hot-server -- --start-hot`, then `webpack-dev-server` will reject it as an invalid argument.

One alternative option is to export a function and pass it in as the `env`, eg:

```
export default function devConfig(env = 'start-hot') {
  return merge(baseConfig, {
    // ...
    devServer: {
      // ...
      setup() {
        if (env === 'start-hot') {
          spawn('npm', ['run', 'start-hot'], { shell: true, env: process.env, stdio: 'inherit' })
            .on('close', code => process.exit(code))
            .on('error', spawnError => console.error(spawnError));
        }
      }
    }
  });
}
```
Thoughts?